src/main/java/jenkins/plugins/git/traits/CloneOptionTrait.java

```diff
@@ -47,6 +49,12 @@ public CloneOptionTrait(CloneOption extension) {
         super(extension);
     }
 
+    @Override
+    protected void decorateContext(SCMSourceContext<?, ?> context) {
+        CloneOption extension = getExtension();
+        ((GitSCMSourceContext<?, ?>) context).shallow(extension.isShallow()).depth(extension.getDepth());
```

Comment:
I think it is safer and clearer to use the Java 17 pattern matching instanceof in this case rather than casting to the correct type.
```suggestion
        if (context instanceof GitSCMSourceContext gitContext) {
            CloneOption extension = getExtension();
            gitContext.shallow(extension.isShallow()).depth(extension.getDepth());
        }
```